src/JMS/Serializer/GraphNavigator.php

```diff
@@ -151,7 +151,12 @@ public function accept($data, array $type = null, Context $context)
                     // metadata for the actual type of the object, not the base class.
                     if (class_exists($type['name'], false) || interface_exists($type['name'], false)) {
                         if (is_subclass_of($data, $type['name'], false)) {
-                            $type = array('name' => get_class($data), 'params' => array());
+                            if (in_array('Doctrine\Common\Proxy\Proxy', class_implements($data), true)) {
```

Comment:
The the only other alternative I see is to iterate over the parents in the event dispatcher and check if there is a listener on them. But I am not sure what impact this can have.